frontend/src/pages/notebookController/StartServerModal.tsx

```diff
@@ -1,28 +1,54 @@
 import * as React from 'react';
-import {
-  ActionList,
-  ActionListItem,
-  Button,
-  Modal,
-  ModalVariant,
-  Spinner,
-} from '@patternfly/react-core';
-import NotebookControllerContext from './NotebookControllerContext';
+import { Alert, AlertVariant, Button, Modal, ModalVariant, Spinner } from '@patternfly/react-core';
+import { Notebook } from '../../types';
+import { checkNotebookRunning } from '../../utilities/notebookControllerUtils';
 
 import './NotebookController.scss';
 
 type StartServerModalProps = {
+  notebook: Notebook | undefined;
   startShown: boolean;
   setStartModalShown: (shown: boolean) => void;
   onClose: () => void;
 };
 
+type SpawnStatus = {
+  status: AlertVariant;
+  title: string;
+  reason: React.ReactNode;
+};
+
 const StartServerModal: React.FC<StartServerModalProps> = ({
+  notebook,
   startShown,
   setStartModalShown,
   onClose,
 }) => {
-  const { notebook, isNotebookRunning } = React.useContext(NotebookControllerContext);
+  const [spawnStatus, setSpawnStatus] = React.useState<SpawnStatus | null>(null);
+
+  const isNotebookRunning = checkNotebookRunning(notebook);
+
+  React.useEffect(() => {
+    let timer;
+    if (isNotebookRunning) {
+      setSpawnStatus({
+        status: AlertVariant.success,
+        title: 'Success',
+        reason: 'The notebook server is up and running. This page will update momentarily.',
+      });
+      timer = setTimeout(() => {
+        if (notebook?.metadata.annotations?.['opendatahub.io/link']) {
+          window.location.href = notebook.metadata.annotations['opendatahub.io/link'];
+        }
+      }, 6000);
+    }
+    return () => {
+      if (timer) {
+        clearTimeout(timer);
+      }
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Don't do this. I know other spots have it, but there is a reason we have a linter telling you about this. This is not a good idea.

If `notebook` is not a stable object reference than pull the link out and use that as a variable. It shouldn't be changing and will be easier for React to compare.

Reply:
Updated.